### DISCUSSION TITLE
Compiler errors when trying to render a @for in a @forelse

### DISCUSSION THREAD

**Discussion by alexwass-lr** [livewire/livewire#7981] [7. Bugs (archived)] [answered]
### Livewire version

v3.4.5

### Laravel version

v10.29.0

### Which PHP version are you using?

PHP 8.2

### Steps To Reproduce

1. Create a new basic component
2. Within the view, add a `@for` loop within a `@forelse` loop
3. You should get the template error `syntax error, unexpected token "endfor", expecting "elseif" or "else" or "endif"`
4. Remove the `@for` loop and the page will render again

### Repository or Wirebox URL

https://wirebox.app/b/4lodg

### Additional information

This seems to be related to PR #7967 @joshhanley.

As you can see in the Wirebox, this is working perfectly fine on v3.4.4. Unfortunately they don't seem to have v3.4.5 available yet for me to create a demonstration on the affected version.

When I strip my local template back to the following, I still get the error in step 3 above (removing the `@for` loop fixes it):
```blade
<div>
    @forelse ($this->assignments as $assignment)
        @for ($i=0; $i < 10; $i++)
        @endfor
    @empty
    @endforelse
</div>
```

Looking closer at the compiled view, here is v3.4.5:
```php
<div>
    <!--[if BLOCK]><![endif]--><?php $__empty_1 = true; $__currentLoopData = $this->assignments; $__env->addLoop($__currentLoopData); foreach($__currentLoopData as $assignment): $__env->incrementLoopIndices(); $loop = $__env->getLastLoop(); $__empty_1 = false; ?>
        <!--[if BLOCK]><![endif]--><?php for($i=0; $i < $assignment->days; $i++): ?>
        <?php endfor; ?><!--[if ENDBLOCK]><![endif]-->
    <?php endforeach; $__env->popLoop(); $loop = $__env->getLastLoop(); if ($__empty_1): ?>
    <?php endfor; ?><!--[if ENDBLOCK]><![endif]-->else
</div>
```

And if I switch to my local copy of Livewire, and checkout to the commit just before that PR, this is what the view looks like:
```php
<div>
    <!--[if BLOCK]><![endif]--><?php $__empty_1 = true; $__currentLoopData = $this->assignments; $__env->addLoop($__currentLoopData); foreach($__currentLoopData as $assignment): $__env->incrementLoopIndices(); $loop = $__env->getLastLoop(); $__empty_1 = false; ?>
        <!--[if BLOCK]><![endif]--><?php for($i=0; $i < $assignment->days; $i++): ?>
        <?php endfor; ?> <!--[if ENDBLOCK]><![endif]-->
    <?php endforeach; $__env->popLoop(); $loop = $__env->getLastLoop(); if ($__empty_1): ?>
    <?php endif; ?> <!--[if ENDBLOCK]><![endif]-->
</div>
```

Here's a diff of the two which highlights the error:
```diff
 <div>
     <!--[if BLOCK]><![endif]--><?php $__empty_1 = true; $__currentLoopData = $this->assignments; $__env->addLoop($__currentLoopData); foreach($__currentLoopData as $assignment): $__env->incrementLoopIndices(); $loop = $__env->getLastLoop(); $__empty_1 = false; ?>
         <!--[if BLOCK]><![endif]--><?php for($i=0; $i < $assignment->days; $i++): ?>
-        <?php endfor; ?> <!--[if ENDBLOCK]><![endif]-->
+        <?php endfor; ?><!--[if ENDBLOCK]><![endif]-->
     <?php endforeach; $__env->popLoop(); $loop = $__env->getLastLoop(); if ($__empty_1): ?>
-    <?php endif; ?> <!--[if ENDBLOCK]><![endif]-->
+    <?php endfor; ?><!--[if ENDBLOCK]><![endif]-->else
 </div>
```

Even more strangely, if I add the following test to `src/Features/SupportMorphAwareIfStatement/UnitTest.php` it still passes.
```php
            28 => [
                2,
                <<<'HTML'
                <div>
                    @forelse([1, 2] as $post)
                        @for($i=0; $i < 10; $i++)
                            <span> {{ $i }} </span>
                        @endfor
                    @empty
                        <span> {{ $someProperty }} </span>
                    @endforelse
                </div>
                HTML
            ],
```

I'll continue investigating this tomorrow to triple confirm this isn't a me issue. Thought I'd give you a heads up in case it wasn't though.

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by joshhanley** [comment, marked as answer]
@alexwass-lr thanks for the detailed report! Yeah I will try and have a look later too. Seems like there might be a bug with the closing statement logic (changing the endfor to an endif). Which the test will still pass as it only asserts the correct number of morph marker blocks.

**Reply by danie-ramdhani** [reply]
@joshhanley it looks like if we use statement like backslash `\\` will returning error. example:

`syntax error, unexpected token "endif"`

```blade
    @foreach (range(1, 5) as $item)
        <span>
            @if ($item > 2 && request()->is(str(url('/'))->replace('\\', '/')))
                foo
            @else
                bar
            @endif
        </span>
    @endforeach
```

or in my case, it will rendered as plain text:

![image](https://github.com/livewire/livewire/assets/58886605/c0492b92-c2f6-4315-8938-55283a570950)

**Reply by joshhanley** [reply]
@danie-ramdhani thanks for letting me know, I will have a look at that too 🙂

**Reply by joshhanley** [reply]
@alexwass-lr and @danie-ramdhani have submitted a PR #7984 that fixes both of your issues 🙂

**Reply by alexwass-lr** [reply]
Amazing, thanks @joshhanley! I can confirm that's sorted my issue

**Reply by joshhanley** [reply]
@alexwass-lr no worries! Thanks for testing
